demo/convolution_demo.py

```diff
@@ -1,41 +1,30 @@
 import pyapr
 
+"""
+This demo reads an APR, applies a convolution operation and displays the result
+"""
 
-def main():
-    """
-    This demo reads an APR, applies a convolution operation and displays the result
-    """
+io_int = pyapr.filegui.InteractiveIO()
+fpath_apr = io_int.get_apr_file_name()  # get APR file path from gui
 
-    io_int = pyapr.filegui.InteractiveIO()
-    fpath_apr = io_int.get_apr_file_name()  # get APR file path from gui
+# Read from APR file
+apr, parts = pyapr.io.read(fpath_apr)
 
-    # Instantiate APR and particle objects
-    apr = pyapr.APR()
-    parts = pyapr.ShortParticles()      # input particles can be float32 or uint16
-    # parts = pyapr.FloatParticles()
+# Stencil and output must be float32
+stencil = pyapr.numerics.filter.get_gaussian_stencil(size=5, sigma=1, ndims=3, normalize=True)
+out = pyapr.FloatParticles()
 
-    # Read from APR file
-    pyapr.io.read(fpath_apr, apr, parts)
+# Convolve using CPU:
+pyapr.numerics.convolve(apr, parts, out, stencil, use_stencil_downsample=True,
+                        normalize_stencil=True, use_reflective_boundary=False)
 
-    # Stencil and output must be float32
-    stencil = pyapr.numerics.filter.get_gaussian_stencil(size=5, sigma=1, ndims=3, normalize=True)
-    out = pyapr.FloatParticles()
+# Alternative CPU convolution method (produces the same result as 'convolve'):
+# pyapr.numerics.convolve_pencil(apr, parts, out, stencil, use_stencil_downsample=True,
+#                                normalize_stencil=True, use_reflective_boundary=False)
 
-    # Convolve using CPU:
-    pyapr.numerics.convolve(apr, parts, out, stencil, use_stencil_downsample=True,
-                            normalize_stencil=True, use_reflective_boundary=False)
+# Convolve using GPU (stencil must be of shape 3x3x3 or 5x5x5):
+# pyapr.numerics.convolve_cuda(apr, parts, out, stencil, use_stencil_downsample=True,
```

Comment:
Removing the gpu version? is this in another demo? or does the above method choose if CUDA is avaliable?

Reply:
These are just code examples. But I can look into adding cuda detection, and then just run both cpu versions and cuda if available. I agree commented out code doesn't look good